File: RESTservice-API/Models/Zone.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace RESTservice_API.Models
{
    public class Zone
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }
        [JsonPropertyName("name")]
        public string Name { get; set; }
        [JsonPropertyName("floorMapId")]
        public int FloorMapId { get; set; }

        [Column(TypeName = "json")]
        [JsonPropertyName("points")]
        public string Points { get; set; } // Serialized array of points
    }
}

File: WebApp/indoor-localization/src/entities/Zone.ts
import { Point } from "./Point";

export class Zone {
    id: number;
    name: string;
    points: Point[];
    parentFacilityId: number;
  
    constructor(
      id: number,
      name: string,
      points: Point[],
      parentFacilityId: number,
    ) {
      this.id = id;
      this.name = name;
      this.points = points;
      this.parentFacilityId = parentFacilityId;
    }
  
  }
  